tests/Functional/AuditingTest.php

```diff
@@ -620,13 +620,16 @@ public function itWillAuditSync()
 
         $article->auditSync('categories', [$secondCategory->getKey()]);
 
+        $audit = $article->audits()->where('event', 'sync')->latest()->first();
         $no_of_audits_after = Audit::where('auditable_type', Article::class)->count();
         $categoryAfter = $article->categories()->first()->getKey();
 
         $this->assertSame($firstCategory->getKey(), $categoryBefore);
         $this->assertSame($secondCategory->getKey(), $categoryAfter);
         $this->assertNotSame($categoryBefore, $categoryAfter);
         $this->assertGreaterThan($no_of_audits_before, $no_of_audits_after);
+        $this->assertCount(1, $audit->old_values['categories']);
+        $this->assertCount(2, $audit->new_values['categories']);
```

Comment:
`sync` is expected to delete the entire relationship and re-attach it

https://laravel.com/docs/11.x/eloquent-relationships#syncing-associations

So after sync, `old_values` has 1, and `new_values` has 1

Reply:
Yeah, I just saw that. I've pushed a fix and a new test which illustrates my problem better. I wasn't expecting you to be this fast 😅
Here's the corresponding issue with more explanation: https://github.com/owen-it/laravel-auditing/issues/931

Reply:
the change I made is how I understood your problem